src/kvstore/raftex/Host.cpp

```diff
@@ -353,8 +381,25 @@ Host::prepareAppendLogRequest() const {
             logs.emplace_back(std::move(le));
         }
         req->set_log_str_list(std::move(logs));
+        req->set_sending_snapshot(false);
     } else {
-        LOG(FATAL) << idStr_ << "We have not support snapshot yet";
+        req->set_sending_snapshot(true);
+        if (!sendingSnapshot_) {
+            LOG(INFO) << idStr_ << "Can't find log " << lastLogIdSent_ + 1
+                      << " in wal, send the snapshot";
+            sendingSnapshot_ = true;
+            part_->snapshot_->sendSnapshot(part_, addr_).then([this] (Status&& status) {
+                if (status.ok()) {
+                    LOG(INFO) << idStr_ << "Send snapshot succeeded!";
+                } else {
+                    LOG(INFO) << idStr_ << "Send snapshot failed!";
+                    // TODO(heng): we should tell the follower i am failed.
+                }
+                sendingSnapshot_ = false;
+            });
+        } else {
+            LOG(INFO) << idStr_ << "The snapshot req is in queue, please wait for a moment";
```

Comment:
Currently, it's not queued but discarded.

Reply:
The task has been putted into event loop inside IOThread in SnapshotManager